packages/nimble-components/src/table/specs/table-column-specs/table-column-duration-text-hld.md

```diff
@@ -0,0 +1,206 @@
+-   APIs to format the duration in a variety of ways
+    -   Show duration as digital display (00:01:20.323)
+    -   Show duration with symbols indicating a quantitative value (i.e. "< 1 sec")
+
+### Risks and Challenges
+
+Formatting the duration units in the exact way that SLE presents them would require custom logic that differs from what the browser provides natively using APIs like `Intl.Numberformat.format()`. The preference here is to align with current browser behaviors for unit formatting, which includes localization.
+
+### Prior Art/Examples
+
+![ElapsedTime SLE](../spec-images/ElapsedTimeSLE.PNG)
+
+## Design
+
+Below is an example of how the `nimble-table-column-duration-text` would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-duration-text field-name="duration">Duration</nimble-table-column-duration-text>
+</nimble-table>
+```
+
+The displayed value for duration will always take the form of "\<days display>, \<hour display>, \<minute display>, \<second display>.\<fractional seconds>", where the commas are non-localized delimiters <sup>1</sup>.
+
+<sup>1.</sup> When and if we are able to adopt the `Intl.DurationFormat` to format values, we will no longer be hard-coding the commas in the string result as that object will produce the entirety of the display text, which may not match what we produce in our implementation.
+
+### API
+
+#### `TableColumnDurationText`
+
+The `TableColumnDurationText` implementation requires no additional API from what `TableColumnTextBase` provides:
+
+```ts
+// inherits field-name attribute
+public class TableColumnDurationText : TableColumnTextBase { }
+
+```
+
+This API would result in a column whose value would only show up to the 'days' unit (i.e. it would show days, hours, minutes, and seconds). We would show up to two fractional digits as needed. Formatting to change slightly from what SLE offers now using "hr" instead of "h", and "sec" instead of "s".
+
+Examples:
+
+| English                | German                    | French             | Chinese                 |
+| ---------------------- | ------------------------- | ------------------ | ----------------------- |
+| "1 hr, 30 min, 2 sec"  | "1 Std., 30 Min., 2 Sek." | "1 h, 30 min, 2 s" | "1 小时, 30 分钟, 2 秒" |
+| "2 hr, 3.55 sec"       | "2 Std., 3,55 Sek."       | "2 h, 3,55 s"      | "2 小时, 3.55 分"       |
+| "3 days, 23 hr, 2 min" | "3 Tg., 23 Std., 2 Min."  | "3 j, 23 h, 2 min" | "3 天, 23 小时, 2 分钟" |
+| "0 sec"                | "0 Sek."                  | "0 s"              | "0 秒"                  |
+
+#### `DurationFormatter`
+
+We will export the class that is responsible for formatting the duration values so that we can use it for formatting values outside of the `Table` (a requirement in SLE).
+
+```ts
+export class DurationFormatter {
+    public constructor(private readonly lang: string) {}
```

Comment:
I guess the only reason I have to push back on this is that the native APIs for `Intl.NumberFormat` and `Intl.DurationFormat` provide the locales as a .ctor parameter much like I have here, and then the rest of the options in a separate parameter.

Reply:
Re-reading the proposal I gave I think it does have some gaps. It would need to take an element reference and also some kind of callback function to fire and notify that the locale has changed.

From that perspective and with your point seems like this would be two separate APIs if comparing to DOM apis. A `DurationFormatter(lang: string, opts)` and a `LangObserver` (similar to ResizeObserver, IntersectionOberserver, etc) which kinda fits in the [ThemeObserver discussion](https://github.com/ni/nimble/issues/783#issuecomment-1287160658).

The `LangObserver` may not be needed today yet since I don't think any Angular apps have different localized subsections or need to respond to lang changes programmatically as they just refresh the page (I don't know if those two assertions are true). If all that's true it could be a future work maybe (assuming such an observer isn't needed for Blazor apps that are localized today)

Reply:
Made a 'Future Considerations' section where I mention the possibility of needing an observer class.